### ISSUE TITLE
[P0] Joint angular–radial challenge: cos(4θ) versus higher harmonics

### ISSUE THREAD

**Issue by LightChainr** [LightChainr/Matching-One#36]
## Motivation

The latest 30-million-replica same-N run finds

```text
A4(N)=N^(13/8) DeltaM / Delta cos(4 theta)
N=65:  0.650 +/- 0.109
N=85:  0.651 +/- 0.136
N=145: 0.557 +/- 0.331
```

with the predicted sign. This is structured evidence for the working `cos(4 theta) N^-13/8` law, but every size currently contains only two orientations. An axis/diagonal-type sign pattern cannot by itself exclude `cos(8 theta)`, `cos(12 theta)`, logarithmic drift, or a finite-size mixture.

A direct four-angle `N=1105` run is an expensive way to answer this. The immediate test should combine many inexpensive two-orientation same-N pairs into one preregistered angular–radial regression.

## Model family

For each same-N pair define

\[
y_N=N^{\alpha}\Delta M_N,
\quad
c_{4,N}=\cos4\theta_1-\cos4\theta_2,
\quad
c_{8,N}=\cos8\theta_1-\cos8\theta_2.
\]

Compare, in this order:

### H4

\[
y_N=A_4c_{4,N}.
\]

### H4 plus radial correction

\[
y_N=c_{4,N}(A_4+B N^{-\omega/2}),
\]

with the candidate `omega` selected on training sizes only.

### H4 plus logarithmic partner

\[
y_N=c_{4,N}(A_4+B\log N).
\]

### H4 + H8

\[
y_N=A_4c_{4,N}+A_8c_{8,N}.
\]

### Free radial exponent

\[
\Delta M_N=A_4c_{4,N}N^{-\alpha},
\]

with `alpha` trained only on the declared training set.

The primary exponent remains frozen at

\[
\alpha=13/8.
\]

## Geometry set

Use primitive Gaussian pairs so every comparison has cyclic quotient `Z/NZ` and the canonical shared-field coupling:

```text
N=65   (8,1)/(7,4)
N=85   (9,2)/(7,6)
N=130  (11,3)/(9,7)
N=145  (12,1)/(9,8)
N=170  (13,1)/(11,7)
N=185  (13,4)/(11,8)
N=221  (14,5)/(11,10)
N=265  (16,3)/(12,11)
```

The frozen confirmation run from PR #33 supplies the first five. Later pairs should be selected for angular leverage and power *before* observing their percolation outputs.

## Train/holdout design

Initial split:

```text
training: N=65,85,130
held out: N=145,170
```

After that score is committed, a second stage may train through `N=170` and hold out a prespecified subset of `185,221,265`.

Do not choose the held-out set by z-score. Use the full batch covariance, including shared-random-field correlations where applicable.

## Why this is stronger than a large single-N angular fit

Different pairs have different `(Delta cos4, Delta cos8)` design vectors. Jointly fitting them can identify whether the observed signs and magnitudes are consistently explained by one spin-4 coefficient, while keeping physical sizes in the range where the signal is measurable.

Only if H4 survives this challenge should the project spend major resources on the `N=1105` leave-one-angle-out experiment.

## Required outputs

- exact rational or high-precision `cos(4 theta)` and `cos(8 theta)` columns;
- per-seed and pooled `Delta M` with batch covariance;
- frozen model definitions and train/holdout split;
- held-out signed residuals and covariance-aware chi-square;
- condition numbers and coefficient correlations;
- leave-one-size-out predictions;
- a clear statement of whether `A8` is resolved, bounded, or unidentifiable.

## Acceptance

Call the leading spin-4 assignment confirmed at this stage only if:

1. H4 predicts held-out signs and magnitudes;
2. its scaled amplitude is stable under an independent seed;
3. H8 or logarithmic alternatives do not obtain an out-of-sample advantage large enough to overturn H4;
4. the conclusion is not driven solely by `N=65,85`;
5. failed candidate models and unstable fits remain committed.

Related: #22, #35, PR #21, PR #33.

**Comment by LightChainr** [comment]
## Geometry-design sanity check

The proposed first five pairs have useful higher-harmonic leverage. Exact evaluation gives approximately

```text
N=65:   DeltaCos8 / DeltaCos4 = +0.788639
N=85:   DeltaCos8 / DeltaCos4 = -0.623945
N=130:  DeltaCos8 / DeltaCos4 = -0.788639
N=145:  DeltaCos8 / DeltaCos4 = -0.054602
N=170:  DeltaCos8 / DeltaCos4 = +0.623945
```

Thus the `cos8` design column is neither proportional to `cos4` nor of one sign. A spin-8 contribution that imitates the observed `N=65` amplitude would make different, sometimes opposite, predictions at `N=85/130/170`. The frozen train/holdout split therefore has genuine angular identifiability even though each individual size supplies only two orientations.

Record the exact rational design values in the analysis artifact rather than recomputing them from rounded angles.

**Comment by LightChainr** [comment]
Frozen H4/H8 angular-radial challenge completed in [`fd1e097`](https://github.com/LightChainr/Matching-One/commit/fd1e097). Exact rational DeltaCos4/DeltaCos8 columns are committed. Training N=65,85,130 and scoring N=145,170 gives held-out chi-square `1.058` for fixed H4 `N^-13/8`, versus `1.100` for H4+H8, `1.712` power, `1.726` log, `1.661` free alpha, and `37.32` zero effect. The H8 coefficient is `-0.0345 +/- 0.0542`: bounded/unresolved, with no out-of-sample advantage over H4.

**Comment by LightChainr** [comment]
## Review of Q43 result on `agent/q42-q43-closure-harmonics` (`ee8860f`)

The frozen split is respected (train N=65,85,130; holdout N=145,170), the exact H4/H8 design is correct, and the central conclusion is conservative: the fitted A8 is in-sample resolved but **not holdout-confirmed**, so it remains bounded rather than promoted.

Three revisions are required before merge:

1. The report says “covariance-aware chi-square”, but the implementation is diagonal WLS using per-size SEs. It omits cross-N batch covariance and training-parameter prediction uncertainty. An independent full-covariance recomputation gives H4 holdout chi-square about **2.19** and H4+H8 about **3.23**, so the scientific winner is unchanged, but the method/label must be corrected.
2. `A_p_from_AM_over_B` in the exported batch-observable table is algebraically wrong (the separately implemented/reported `A_p_pred` is correct, so the report conclusion is unaffected).
3. LOSO reuses omega/alpha selected on the full training set; call it frozen-hyperparameter LOSO or nest the hyperparameter selection.

The commit also depends on the unmerged C05 parent and cannot run standalone on the current PR #21 head. Decision: retain the branch and conclusions as reviewed evidence, but do not cherry-pick until the parent is corrected and these three items are revised.

**Comment by LightChainr** [comment]
## Stage-1 result and frozen Stage-2 design

Checkpoint `fd1e097` completes the initial preregistered challenge:

- training sizes: `N=65,85,130`;
- held out: `N=145,170`;
- fixed `alpha=13/8`, H4 model held-out chi-square: `1.058` for three retained seed rows;
- zero-effect benchmark: `37.32`;
- H4+H8: `1.100`, with `A8=-0.0345 +/- 0.0542`;
- power, logarithmic and free-alpha alternatives all score worse out of sample.

This is a successful Stage-1 held-out test of the structural H4 law. It favors the preregistered `DeltaCos4*N^-13/8` model over zero and over the declared alternatives at current precision. It still does not sharply determine the asymptotic exponent, because the alternatives are weakly identified and several have large condition numbers.

The clean-checkout replay and provenance/RNG requirements in #39 remain mandatory before final closure.

### Freeze Stage 2 now

Use the already declared primitive pairs:

```text
N=185  (13,4)/(11,8)   new training extension
N=221  (14,5)/(11,10)  held out
N=265  (16,3)/(12,11)  held out
```

After the clean replay, fit through `N=185` and reveal `N=221,265` only for scoring. Keep the model order unchanged:

1. fixed `13/8` H4;
2. fixed `13/8` H4 plus one training-selected power correction;
3. fixed `13/8` H4 plus logarithmic amplitude;
4. fixed `13/8` H4+H8;
5. free-alpha H4 trained only on the declared training set.

The expected effects at these sizes are only of order `1e-4`. Power sample counts from the existing 100-million covariance; a target near `4e-5` standard error will likely require roughly `4e8--5e8` paired replicas per new size. Freeze the final count before viewing any new percolation output.

Required additions:

- full clean-source metadata from #39;
- explicit cross-size RNG/covariance policy;
- per-seed reporting before pooling;
- signed held-out residuals and covariance-aware chi-square;
- condition numbers and H8 confidence interval;
- no `N=1105` campaign unless the two new held-out sizes retain the H4 advantage.

Stage 1 is passed; keep this issue open for Stage 2 and the final harmonic/exponent boundary.

**Comment by LightChainr** [comment]
## P37 no-fit doubling result passes; add H12 before Stage 2

A fresh 100-million-replica seed has now scored the preregistered exact doubling relation

```text
Delta M(2N) / Delta M(N) = -2^(-13/8)
```

without fitting an amplitude or exponent:

```text
65 -> 130: ratio -0.31382 +/- 0.0908; residual z=+0.114
85 -> 170: ratio -0.34095 +/- 0.1118; residual z=-0.150
joint fixed-residual chi2 = 0.03445 / 2
```

The protocol was committed at `ceb7a578...` before the fresh run; the run uses clean source commit `80fbdd1...`, new seed `2026100101`, counters `[2,000,000,000,2,100,000,000)`, 100 batches and all four sizes.

This strengthens Stage 1 substantially: one operation fixes both the scale ratio and the angular phase. However, rotation by `pi/4` acts on square-lattice harmonics as

```text
cos(4 m (theta+pi/4)) = (-1)^m cos(4 m theta).
```

The sign flip therefore distinguishes odd-`m` harmonics (`cos4`, `cos12`, ...) from even-`m` harmonics (`cos8`, `cos16`, ...); it does not uniquely distinguish H4 from H12.

### Freeze one additional Stage-2 alternative

Before any `N=185,221,265` output is viewed, add

```text
fixed_13_8_H4_H12:
  DeltaM = N^(-13/8) * (A4*DeltaCos4 + A12*DeltaCos12)
```

with the same training/held-out split and conditioning rules as H4+H8. `DeltaCos12` is exact from the Gaussian representations and must be committed as a rational/high-precision design column.

The Stage-2 primary remains fixed H4. H12 is a falsification alternative, not a post-hoc rescue. Keep H8 as well because the wider angular design should independently bound both parity classes.

### Full-curve continuation

Issue #49 now freezes the cleaner threshold-rank version at `u=0,0.025,0.050`, plus the induced no-fit root-gap ratio `-1/4`. The `N=1105` campaign remains gated on Stage 2 and #49.

**Comment by LightChainr** [comment]
## Required model-set update after the completed doubling test

The `1+i` result excludes dominance by even-`m` square harmonics but does not distinguish `cos4` from `cos12`, because both change sign under `theta -> theta+pi/4`.

Please add an explicit H12 column/model to the next frozen angular challenge:

```text
y_N = A4 DeltaCos4 + A12 DeltaCos12
```

with `DeltaCos12` generated by exact Chebyshev/rational arithmetic. Score pure H4 before the H4+H12 extension.

Issue #57 supplies a more direct parameter-free discriminator using norm-5 Gaussian multipliers. It predicts opposite child signs for H4 and H12 at `N=325,425`; PR #56 contains the frozen artifact and exact generator. The expensive `N=1105` fit should remain downstream of this cheaper test.

**Comment by LightChainr** [comment]
结案当前阶段：冻结的 H4 / radial-correction / log / H4+H8 / free-alpha challenge 已经按 training `N=65,85,130`、held-out `N=145,170` 跑完，固定 `13/8` H4 在该设计上 held-out 表现最好，H8 未解析。后来识别出的关键剩余歧义是更高 odd harmonics（尤其 H12），已经由更有判别力的 prospective #55/#57 接管。这个 issue 不再扩成无限模型列表。
